benchmark_runner/grafana/update_grafana_dashboard.py

```diff
@@ -0,0 +1,24 @@
+
+from benchmark_runner.common.grafana.grafana_operations import GrafanaOperations
+from benchmark_runner.common.grafana.update_grafana_last_value_mappings import UpdateGrafanaLastValueMappings
+from benchmark_runner.main.environment_variables import environment_variables
+
+# Update perf main.libsonnet with last versions
+update_grafana_mappings_value = UpdateGrafanaLastValueMappings(main_libsonnet_path='perf/jsonnet/main.libsonnet')
+last_versions = update_grafana_mappings_value.get_last_elasticsearch_versions()
+update_grafana_mappings_value.update_value_mappings_last_versions(last_versions=last_versions)
+update_grafana_mappings_value.update_main_libsonnet()
+
+
+# Update grafana perf dashboard
+environment_variables_dict = environment_variables.environment_variables_dict
+grafana_operations = GrafanaOperations(grafana_url=environment_variables_dict.get('grafana_url', ''),
+                                       grafana_api_key=environment_variables_dict.get('grafana_api_key', ''),
+                                       grafana_json=environment_variables_dict.get('grafana_json', ''))
+# for debug: grafana_operations.fetch_all_dashboards()
+grafana_operations.read_dashboard_json()
+grafana_operations.increment_dashboard_version()
+grafana_operations.override_dashboard()
+
+# Error: 412
+# The 412 status code is used when a newer dashboard already exists (newer, its version is greater than the version that was sent). The same status code is also used if another dashboard exists with the same title.
```

Comment:
What does this comment refer to (this looks like the entire file)?

Reply:
If We got this error, to explain the reason

Reply:
It's just a comment at the very end of a file.  This comment should go with whatever can actually produce that error, or better yet, if the error is received, there should be an error message.

Reply:
You right, Move the error code message it to grafana_operation.py where it raised